sihl/src/sihl.mli

```diff
@@ -1,9 +1,180 @@
+      [root_path] is used. If no root path can be found, [None] is returned. *)
+
+  val env_files_path : unit -> string option
+
+  (** [read_env_file ()] reads an [.env] file from the directory given by
+      [env_files_path] and returns the key-value pairs as [data]. If [SIHL_ENV]
+      is set to [test], [.env.test] is read. Otherwise [.env] is read. If the
+      file doesn't exist or the directory containing the file can't be found,
+      [None] is returned.
+
+      If you just want to access configuration values, use the read functions
+      instead. Every time you call [read_env_file] the file is read from disk. *)
+  val read_env_file : unit -> data option
+
+  (** [load_env_file ()] reads an [.env] file using {!read_env_file} and stores
+      its contents into the environment variables. *)
+  val load_env_file : unit -> unit
+
+  (** [load ()] calls {!load_env_file} and makes sure that [SIHL_ENV] was set. *)
+  val load : unit -> unit
+
+  (** [read schema] returns the decoded, statically typed version of
+      configuration [t] of the [schema]. This is used in services to
+      declaratively define a valid configuration.
+
+      The configuration data [t] is merged with the environment variable and, if
+      present, an .env file.
+
+      It fails with [Exception] and prints descriptive message of invalid
+      configuration. *)
+  val read : ('ctor, 'ty) schema -> 'ty
+
+  (** [read_string key] returns the configuration value with [key] if present.
+      The function is memoized, the first call caches the returned value and
+      subsequent calls are fast. *)
+  val read_string : string -> string option
+
+  (** [read_secret unit] returns the value of SIHL_SECRET if it is set. If
+      SIHL_SECRET was not set, it fails in production and in testing or local
+      development the value is set to "secret". *)
+  val read_secret : unit -> string
+
+  (** [read_int key] returns the configuration value with [key] if present. the
+      first call caches the returned value and subsequent calls are fast. *)
+  val read_int : string -> int option
+
+  (** [read_bool key] returns the configuration value with [key] if present. the
+      first call caches the returned value and subsequent calls are fast. *)
+  val read_bool : string -> bool option
+
+  (** [is_test ()] returns true if Sihl is running in a test environment,
+      meaning if tests are executing parts of Sihl. *)
+  val is_test : unit -> bool
+
+  (** [is_development ()] returns true if Sihl is running in a local
+      environment. *)
+  val is_development : unit -> bool
+
+  (** [is_production ()] returns true if Sihl is running in a production
+      environment. *)
+  val is_production : unit -> bool
+
+  (** [require t] raises an exception if the stored configuration doesn't
+      contain the configurations provided by the list of configurations [t]. *)
+  val require : ('ctor, 'ty) schema -> unit
```

Comment:
Could you move this up to `read`? It seems like this should be grouped with that.

Reply:
Fixed.